pkg/image/api/helper.go

```diff
@@ -400,8 +401,42 @@ func ImageWithMetadata(image *Image) error {
 			image.DockerImageMetadata.Size = v1Metadata.Size
 		}
 	case 2:
-		// TODO: need to prepare for this
-		return fmt.Errorf("unrecognized Docker image manifest schema %d for %q (%s)", manifest.SchemaVersion, image.Name, image.DockerImageReference)
+		config := DockerConfigImage{}
+		if err := json.Unmarshal([]byte(image.DockerConfigImage), &config); err != nil {
+			return fmt.Errorf("JSON error (1): %v: <%s>", err, image.DockerConfigImage)
+		}
+
+		image.DockerImageLayers = make([]ImageLayer, len(manifest.Layers))
+		for i, layer := range manifest.Layers {
+			image.DockerImageLayers[i].MediaType = layer.MediaType
+			image.DockerImageLayers[i].Name = layer.Digest
+			image.DockerImageLayers[i].Size = layer.Size
+		}
+		// reverse order of the layers for v1 (lowest = 0, highest = i)
+		for i, j := 0, len(image.DockerImageLayers)-1; i < j; i, j = i+1, j-1 {
+			image.DockerImageLayers[i], image.DockerImageLayers[j] = image.DockerImageLayers[j], image.DockerImageLayers[i]
+		}
+
+		image.DockerImageMetadata.ID = manifest.Config.Digest
+		image.DockerImageMetadata.Parent = config.Parent
+		image.DockerImageMetadata.Comment = config.Comment
+		image.DockerImageMetadata.Created = config.Created
+		image.DockerImageMetadata.Container = config.Container
+		image.DockerImageMetadata.ContainerConfig = config.ContainerConfig
+		image.DockerImageMetadata.DockerVersion = config.DockerVersion
+		image.DockerImageMetadata.Author = config.Author
+		image.DockerImageMetadata.Config = config.Config
+		image.DockerImageMetadata.Architecture = config.Architecture
+
+		if len(image.DockerImageLayers) > 0 {
+			size := int64(0)
+			for _, layer := range image.DockerImageLayers {
+				size += layer.Size
+			}
+			image.DockerImageMetadata.Size = size
```

Comment:
But the `size` is initialized to 0. I believe https://github.com/openshift/origin/pull/8938/commits/fddfda1cb66ef9decabf2e4f838bc0803ec913a9#diff-78bcdfab573a18c6ae2ac4e175b48addR430 should follow this block.